src/main/java/com/digirati/taxonomy/manager/normalisation/TextNormaliser.java

```diff
@@ -0,0 +1,74 @@
+package com.digirati.taxonomy.manager.normalisation;
+
+import edu.stanford.nlp.ling.CoreLabel;
+import edu.stanford.nlp.pipeline.CoreDocument;
+import edu.stanford.nlp.pipeline.StanfordCoreNLP;
+
+import java.util.Collection;
+import java.util.Properties;
+
+/**
+ * Provides a mechanism for normalising any text inputs, including the terms to search for and the
+ * text in which to search.
+ */
+public class TextNormaliser {
+
+    private final Collection<String> stopwords;
+
+    private final StanfordCoreNLP pipeline;
+
+    /**
+     * Constructs a {@link TextNormaliser} capable of correctly lemmatising English text
+     *
+     * @param stopwords a collection of common boilerplate words to remove from the text (e.g.
+     *     "the", "a", etc.)
+     */
+    public TextNormaliser(Collection<String> stopwords) {
+        this(stopwords, "en", "English");
+    }
+
+    /**
+     * Constructs a {@link TextNormaliser} for normalising any input language. Currently, full
+     * lemmatisation is only supported for English due to a restriction of the NLP library that
+     * we're using, but language-specific tokenisation should still work
+     *
+     * @param stopwords a collection of common boilerplate words to remove from the text (e.g.
+     *     "the", "a", etc.)
+     * @param languageKey the ISO 639-1 language code for the text to be normalised
+     * @param languageName the name of the language of the text to be normalised
+     */
+    public TextNormaliser(Collection<String> stopwords, String languageKey, String languageName) {
+        this.stopwords = stopwords;
+        this.pipeline = createPipeline(languageKey, languageName);
```

Comment:
This call-chain (`createPipeline`) does an expensive set up of the annotation pipeline. This needs to happen somewhere where the computational cost is obvious. The expectation of the `new` operator is that it only allocates for the object you're creating.